nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -69,7 +69,8 @@ fi
 chown -f 0:30000 /nix/store
 chmod -f 1775 /nix/store
 if [ -n "@readOnlyNixStore@" ]; then
-    if ! [[ "$(findmnt --noheadings --output OPTIONS /nix/store)" =~ ro(,|$) ]]; then
+    # #375257: Ensure that we pick the "top" (i.e. last) mount so we don't get a false positive for a lower mount.
+    if ! [[ "$(findmnt --noheadings --output OPTIONS /nix/store | tail -n1)" =~ (^|,)ro(,|$) ]]; then
```

Comment:
Is this equivalent?

```suggestion
    if ! [[ "$(findmnt --first-only --noheadings --output OPTIONS /nix/store)" =~ (^|,)ro(,|$) ]]; then
```

Reply:
It isn't, in a case where (for instance) there is a RW backing layer mounted under a RO overlay, shadowing the RW mount. Consider this case from the test script:

```
machine # /dev/vdb on /nix/store type ext4 (rw,relatime,nodioread_nolock,nodelalloc,errors=remount-ro,data=journal)
machine # nix-store on /nix/store/ro type 9p (rw,relatime,cache=f,access=client,msize=16384,trans=virtio)
machine # overlay on /nix/store type overlay (rw,relatime,lowerdir=/mnt-root/nix/store/ro,upperdir=/mnt-root/nix/store/rw,workdir=/mnt-root/nix/store/work,uuid=on)
```

--first-only picks the /dev/vdb rw mount instead of the overlay. If that overlay were already readonly we would not want to create the bind mount on top of it.